app/models/stripe_record.rb

```diff
@@ -36,28 +47,42 @@ class StripeRecord < ApplicationRecord
   # Also saves us from some pains because JSON columns are highly inconsistent among MySQL and MariaDB.
   serialize :parameters, coder: JSON
 
+  def determine_wca_status
+    result = WCA_TO_STRIPE_STATUS_MAP.find { |key, values| values.include?(stripe_status) }
+    return result.first unless result.nil?
```

Comment:
`result&.first || raise "Foo"` _should_ do the same thing, although I'm writing this from the top of my head so you should quickly test it